### PR TITLE
Dev sir 1332 user context

### PR DESCRIPTION
**Pull request by OttoVT**

### CODE REVIEW

File: src/Api.Common/Persistence/WithdrawalRepository.cs [hunk 1]
```diff
@@ -395,7 +395,7 @@ private WithdrawalUpdate CreateUpdate(Withdrawal withdrawal)
                 AccountId = withdrawal.AccountId,
                 BrokerAccountDetails = withdrawal.BrokerAccountDetails,
                 State = withdrawal.State,
-                ReferenceId = withdrawal.ReferenceId,
+                AccountReferenceId = withdrawal.AccountReferenceId,
```
Comment (KonstantinRyazantsev) [inline src/Api.Common/Persistence/WithdrawalRepository.cs:398]: Where is user context?

File: src/Api.Common/Persistence/WithdrawalRepository.cs [hunk 2]
```diff
@@ -83,7 +83,7 @@ public async Task<WithdrawalUpdate> Upsert(Withdrawal withdrawal)
                             State = withdrawal.State,
                             BrokerAccountId = withdrawal.BrokerAccountId,
                             Sequence = withdrawal.Sequence,
-                            ReferenceId = withdrawal.ReferenceId,
+                            AccountReferenceId = withdrawal.AccountReferenceId,
```
Comment (KonstantinRyazantsev) [inline src/Api.Common/Persistence/WithdrawalRepository.cs:86]: Where is user context?

File: src/Api.Worker/Api.Worker.csproj [hunk 3]
```diff
@@ -23,7 +23,7 @@
     <PackageReference Include="MassTransit.RabbitMQ" Version="6.3.2" />
     <PackageReference Include="swisschain.extensions.Postgres" Version="1.0.0" />
     <PackageReference Include="Swisschain.Sdk.Server" Version="6.8.0" />
-    <PackageReference Include="Swisschain.Sirius.Brokerage.MessagingContract" Version="3.6.0" />
+    <PackageReference Include="Swisschain.Sirius.Brokerage.MessagingContract" Version="3.7.0-beta.2" />
```
Comment (KonstantinRyazantsev) [inline src/Api.Worker/Api.Worker.csproj:26]: release it

File: src/Api/Api.csproj [hunk 4]
```diff
@@ -31,7 +31,7 @@
     <PackageReference Include="Microsoft.Extensions.Caching.StackExchangeRedis" Version="3.1.7" />
     <PackageReference Include="Swisschain.Extensions.Grpc.Abstractions" Version="2.0.0" />
     <PackageReference Include="Swisschain.Sdk.Server" Version="6.8.0" />
-    <PackageReference Include="Swisschain.Sirius.Brokerage.ApiClient" Version="2.5.0" />
+    <PackageReference Include="Swisschain.Sirius.Brokerage.ApiClient" Version="2.6.0-beta.2" />
```
Comment (KonstantinRyazantsev) [inline src/Api/Api.csproj:34]: Release it

File: src/Api/GrpcServices/WithdrawalService.cs [hunk 5]
```diff
@@ -237,12 +239,19 @@ public override async Task<WithdrawalExecuteResponse> Execute(WithdrawalExecuteR
                         $"Shortage of {shortage} {asset.Symbol}");
                 }
 
+                var userContext = new UserContext()
```
Comment (KonstantinRyazantsev) [inline src/Api/GrpcServices/WithdrawalService.cs:242]: why it's not passed somewhere?

File: src/Api/WebApi/Models/Withdrawals/WithdrawalsSearchRequest.cs [hunk 6]
```diff
@@ -17,7 +17,7 @@ public class WithdrawalsSearchRequest : PaginationRequest<long?>
         public long? AccountId { get; set; }
         
         [FromQuery(Name = "referenceId")]
```
Comment (KonstantinRyazantsev) [inline src/Api/WebApi/Models/Withdrawals/WithdrawalsSearchRequest.cs:19]: rename this too

Comment (KonstantinRyazantsev) [reply src/Api/WebApi/Models/Withdrawals/WithdrawalsSearchRequest.cs:19]: and don't forget about UI

File: src/Api/GrpcServices/WithdrawalService.cs [hunk 7]
```diff
@@ -411,7 +412,16 @@ response.Response.Error.ErrorCode switch
...
                     AssetSymbol = asset.Symbol,
-                    BrokerAccountName = brokerAccount.Name
+                    BrokerAccountName = brokerAccount.Name,
+                    UserContext = new Common.ReadModels.Withdrawals.UserContext()
+                    {
+                        AccountReferenceId = response.Response.UserContext.AccountReferenceId,
+                        WithdrawalReferenceId = response.Response.UserContext.WithdrawalReferenceId,
+                        UserId = response.Response.UserContext.UserId,
+                        ApiKeyId = response.Response.UserContext.ApiKeyId,
+                        PassClientIp = response.Response.UserContext.PassClientIp,
```
Comment (KonstantinRyazantsev) [inline src/Api/GrpcServices/WithdrawalService.cs:422]: What is "Pass"?

File: src/Api.ApiContract/common.proto [hunk 8]
```diff
@@ -130,4 +130,5 @@ message TransactionInfo
 message UserContext{
   google.protobuf.StringValue account_reference_id = 4;
   google.protobuf.StringValue withdrawal_reference_id = 5;
+  google.protobuf.StringValue withdrawal_params_signature = 6;
```
[the remark this answers is not in the document]
Comment (KonstantinRyazantsev) [reply src/Api.ApiContract/common.proto:133]: user's context it's something related to the user's system, used by our system just for tracing purposes, and can be used by the transaction validation policy written by the user. The signature is another thing.

File: src/Api/GrpcServices/WithdrawalService.cs (hunk 7 above)
Comment (OttoVT) [reply src/Api/GrpcServices/WithdrawalService.cs:422]: It was a name of the ip field in the task https://swisschain.atlassian.net/browse/SIR-1243

File: src/Api.ApiContract/common.proto (hunk 8 above)
[the remark this answers is not in the document]
Comment (OttoVT) [reply src/Api.ApiContract/common.proto:133]: It should be part of a user context. Alexey told me to include it in user context.

File: src/Api/GrpcServices/WithdrawalService.cs (hunk 7 above)
Comment (KonstantinRyazantsev) [reply src/Api/GrpcServices/WithdrawalService.cs:423]: No, there is a separate word "pass" which means that client IP should be passed :)

**Review by KonstantinRyazantsev — APPROVED** [review]

**Pull request merged**